### PR TITLE
Feature/postgresql full text search

### PR DESCRIPTION
**Pull request by juuso-j**
# Search endpoint for the smbackend.
Search endpoint that uses postgresql full text search. Also updates the backend to django4 and python 3.10.

### Documentation

1. README.md
    * Updated/Corrected the part of installing the smbackend. Now instructions that should work, when setting up the project and the database.
2. config_dev.env.example
    * Removed elasticsearch and haystack examples.
3. services/search/specification.swagger.yaml
    * Swagger documentation for the search endpoint.
4. smbackend_turku/README.md
    * Updated with indexing info.     
### Api
1. services/api.py
    * Removed SearchViewSet and haystack code.
2. services/search/api.py
    * Contains the endpoint that parses the parameters, makes the search query, orderings and optional filterings. Includes also search results serializers for all model types. 

### Migrations, models and indexing
1. services/apps.py
    * Signal registering.
2. services/management/commands/index_search_columns.py
    * Management script that creates indexes to the search_columns.
3. services/migrations/0087_create_search_columns.py
    * Create Gindex search_columns for Unit and Service models.
4. services/migrations/0088_create_search_view.py
    * Raw sql migration that creates a view from search_columns in Unit, Service, Address and Administrative models.
5. services/models/service.py
    * Added search_column as Gin and get_search_column_indexing function that return the indexing configuration.
6. services/models/unit.py
   * Added search_column as Gin and get_search_column_indexing function that return the indexing configuration.
7. services/signals.py
    * Post_save signals form the models that are indexed for searching. Updates the search_column when model is saved.
8. smbackend_turku/importers/addresses.py
    * Populates the multilingual full_name column which is used when indexing addresses to the search_column
9. smbackend_turku/importers/units.py
    * Adds 1 (self produced) to all units imported from the MDS, in search_results, result are ordered by the provider type. This allows to set a different provider_type to units imported from external sources and give them less weight when ordering in results.   

### Tests
1. services/search/tests/fixtures.py
   * Fixture data for the tests.
2. services/search/tests/test_api.py
   * Basic API test for searching for Units, Addresses, Services and Administrative divisions.
   * 
### Removal of haystack and elasticsearch
1. multilingual_haystack/__init__.py
2.  multilingual_haystack/backends.py
3. multilingual_haystack/custom_elasticsearch_search_backend.py
4. services/management/commands/verify_search_index_integrity.py
5. services/search_indexes.py
6. services/search_suggestions.py
7. smbackend/elasticsearch/mappings_english.json
8. smbackend/elasticsearch/mappings_finnish.json
9. smbackend/elasticsearch/mappings_swedish.json
10. smbackend/elasticsearch/settings_english.json
11. smbackend/elasticsearch/settings_finnish.json
12. smbackend/elasticsearch/settings_swedish.json
13. smbackend/settings_test.py
All above files deleted, thus haystack and elasticsearch are no longer used.
    
### Other
1. deploy/Dockerfile_APM
    * Removed elasticsearch.
2. docker-compose.yml
    * Removed elasticsearch.
3. eco_counter/migrations/0003_alter_importstate_current_year_number_and_more.py
    * Migration that for some reason has not been added to the develop and should be there.  
4. observations/migrations/0009_alter_observation_polymorphic_ctype.py
    * Migration created after django polymorphic was updated to a version required by Django4.
5. requirements.in
    * Changed django to version 4, munigeo to version 2.65(this version has search_columns that can be indexed for the search). Django-polymorphic to >3..1 which required by Django4, removed haystack, 
6. requirements.txt
    * Generated from requirements.in with pip-compile.
7. smbackend/settings.py
    * Removed elasticsearch and haystack settings. Added logger for search. Removed hardcoded DB settings.
8. smbackend/urls.py
    * Added url for the new search, removed suggestions url. Removed django.conf.urls.url as it is removed in Django4.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,48 +17,98 @@ First configure development environment settings as stated in `config_dev.env.ex
...
+ ImportError: cannot import name 'html5lib' from 'pip._vendor' (/home/johndoe/.virtualenvs/servicemap/lib/python3.10/site-packages/pip/_vendor/__init__.py)
+```
+Try installing latest pip. 
+```
+curl -sS https://bootstrap.pypa.io/get-pip.py | python3.10
+```
+
+4. Setup the PostGIS database.
 
+Please note, we reqommend PostgreSQL version 13 or higher.
```
Comment (ezkat) [inline README.md:78]: Typo 'reqommend'

**Review by aceViilee — APPROVED** [review]

File: config_dev.env.example [hunk 2]
```diff
@@ -24,10 +24,7 @@
 DEBUG=True
 
 # Languages that this smbackend instance will support. This affects
-# several things and cannot be changed nilly-willy. You will also need
-# to rebuild your elasticsearch indexes: `manage.py rebuild_indexes`,
-# but only if you are actually running elasticsearch, ie. specifying
-# ELASTICSEARCH_URL
+# several things and cannot be changed nilly-willy.
```
Comment (ezkat) [inline config_dev.env.example:27]: 'willy-nilly'

File: README.md (hunk 1 above)
Comment (juuso-j) [reply README.md]: fixed

File: services/api.py [hunk 3]
```diff
@@ -119,7 +118,7 @@ def to_internal_value(self, data):
         }
         Transforms this:
         {
-            "name": "musiikkiklubit",
+            "name": "musiikkiklubit",f
```
Comment (ezkat) [inline services/api.py:121]: f?

File: services/api.py [hunk 4]
```diff
@@ -706,13 +705,13 @@ def to_representation(self, obj):
         if qparams.get("geometry", "").lower() in ("true", "1"):
             geom = obj.geometry  # TODO: different geom types
             if geom and obj.geometry != obj.location:
+                ()
```
Comment (ezkat) [inline services/api.py:708]: empty tuple round brackets?

File: config_dev.env.example (hunk 2 above)
Comment (juuso-j) [reply config_dev.env.example]: Fixed, but that nilly-willy expression has been there before I made my editions. Maybe the previous editor had some meaning to nilly-willy.

File: services/api.py (hunk 3 above)
Comment (juuso-j) [reply services/api.py]: fixed

File: services/api.py (hunk 4 above)
Comment (juuso-j) [reply services/api.py]: Mysterious,  "()", but fixed.

**Review by ezkat — APPROVED** [review]

**Review by Anthon98 — APPROVED** [review]

**Review by pelmat — APPROVED** [review]

**Pull request merged**